avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -189,13 +194,29 @@ protected GenericContainerWithVersion deserializeWithSchemaAndVersion(
       boolean writerSchemaIsPrimitive =
               AvroSchemaUtils.getPrimitiveSchemas().values().contains(writerSchema);
       if (writerSchemaIsPrimitive) {
-        return new GenericDatumReader<>(writerSchema, finalReaderSchema);
+        GenericData genericData = new GenericData();
+        if (avroUseLogicalTypeConverters) {
+          genericData.addLogicalTypeConversion(new Conversions.DecimalConversion());
+        }
+        return new GenericDatumReader<>(writerSchema, finalReaderSchema, genericData);
       } else if (useSchemaReflection) {
-        return new ReflectDatumReader<>(writerSchema, finalReaderSchema);
+        ReflectData reflectData = new ReflectData();
+        if (avroUseLogicalTypeConverters) {
+          reflectData.addLogicalTypeConversion(new Conversions.DecimalConversion());
```

Comment:
Could you also add `TimeConversions.DateConversion`, `TimeConversions.LocalTimestampMillisConversion`, and `TimeConversions.TimeMillisConversion`?   Those are the 3 other logical types supported by the converter.

Perhaps extract a new method `addLogicalTypes(GenericData)` that adds the 4 logical types.

Reply:
So, I added them all. I feel like what's available for kafka connect, is not relevant for people who uses this outside of the kafka-connect framework. If you have have objections I can easily remove the extra ones.